### PR TITLE
[mysql] Add OAuth2 framework tables

### LINKED ISSUE #3550: [REQ][MySQL] Tables for OAuth2 security schema
### Is your feature request related to a problem? Please describe.
While thinking about OAuth2 library for PHP Slim generator in #3549 I've found OAuth database schema in source code. [Here](https://github.com/dsquier/oauth2-server-php-mysql/blob/master/oauth.ddl) and [here](https://github.com/bshaffer/oauth2-server-php/blob/5a0c8000d4763b276919e2106f54eddda6bc50fa/src/OAuth2/Storage/Pdo.php#L649-L727).

## Describe the solution you'd like
It would be great to add these tables to MySQL schema generator. Database schema copied from [oauth2-server-php-mysql](https://github.com/dsquier/oauth2-server-php-mysql/) repo described below.
### oauth_access_tokens
```
+--------------+---------------+------+-----+---------+-------+
| Field        | Type          | Null | Key | Default | Extra |
+--------------+---------------+------+-----+---------+-------+
| access_token | varchar(40)   | NO   | PRI | NULL    |       |
| client_id    | varchar(80)   | YES  |     | NULL    |       |
| user_id      | varchar(80)   | YES  |     | NULL    |       |
| expires      | timestamp     | NO   |     | NULL    |       |
| scope        | varchar(4000) | YES  |     | NULL    |       |
+--------------+---------------+------+-----+---------+-------+
```
### oauth_authorization_codes
```
+--------------------+---------------+------+-----+---------+-------+
| Field              | Type          | Null | Key | Default | Extra |
+--------------------+---------------+------+-----+---------+-------+
| authorization_code | varchar(40)   | NO   | PRI | NULL    |       |
| client_id          | varchar(80)   | YES  |     | NULL    |       |
| user_id            | varchar(80)   | YES  |     | NULL    |       |
| redirect_uri       | varchar(2000) | NO   |     | NULL    |       |
| expires            | timestamp     | NO   |     | NULL    |       |
| scope              | varchar(4000) | YES  |     | NULL    |       |
| id_token           | varchar(1000) | YES  |     | NULL    |       |
+--------------------+---------------+------+-----+---------+-------+
```
### oauth_clients
```
+---------------+---------------+------+-----+---------+-------+
| Field         | Type          | Null | Key | Default | Extra |
+---------------+---------------+------+-----+---------+-------+
| client_id     | varchar(80)   | NO   | PRI | NULL    |       |
| client_secret | varchar(80)   | YES  |     | NULL    |       |
| redirect_uri  | varchar(2000) | YES  |     | NULL    |       |
| grant_types   | varchar(80)   | YES  |     | NULL    |       |
| scope         | varchar(4000) | YES  |     | NULL    |       |
| user_id       | varchar(80)   | YES  |     | NULL    |       |
+---------------+---------------+------+-----+---------+-------+
```
### oauth_jti
```
+----------+---------------+------+-----+---------+-------+
| Field    | Type          | Null | Key | Default | Extra |
+----------+---------------+------+-----+---------+-------+
| issuer   | varchar(80)   | NO   |     | NULL    |       |
| subject  | varchar(80)   | YES  |     | NULL    |       |
| audience | varchar(80)   | YES  |     | NULL    |       |
| expires  | timestamp     | NO   |     | NULL    |       |
| jti      | varchar(2000) | NO   |     | NULL    |       |
+----------+---------------+------+-----+---------+-------+
```
### oauth_jwt
```
+------------+---------------+------+-----+---------+-------+
| Field      | Type          | Null | Key | Default | Extra |
+------------+---------------+------+-----+---------+-------+
| client_id  | varchar(80)   | NO   |     | NULL    |       |
| subject    | varchar(80)   | YES  |     | NULL    |       |
| public_key | varchar(2000) | NO   |     | NULL    |       |
+------------+---------------+------+-----+---------+-------+
```
### oauth_public_keys
```
+----------------------+---------------+------+-----+---------+-------+
| Field                | Type          | Null | Key | Default | Extra |
+----------------------+---------------+------+-----+---------+-------+
| client_id            | varchar(80)   | YES  |     | NULL    |       |
| public_key           | varchar(2000) | YES  |     | NULL    |       |
| private_key          | varchar(2000) | YES  |     | NULL    |       |
| encryption_algorithm | varchar(100)  | YES  |     | RS256   |       |
+----------------------+---------------+------+-----+---------+-------+
```
### oauth_refresh_tokens
```
+---------------+------------------+------+-----+-------------------+-----------------------------+
| Field         | Type             | Null | Key | Default           | Extra                       |
+---------------+------------------+------+-----+-------------------+-----------------------------+
| refresh_token | varchar(40)      | NO   | PRI | NULL              |                             |
| client_id     | varchar(80)      | YES  |     | NULL              |                             |
| user_id       | varchar(80)      | YES  |     | NULL              |                             |
| expires       | timestamp        | NO   |     | CURRENT_TIMESTAMP | on update CURRENT_TIMESTAMP |
| scope         | varchar(4000)    | YES  |     | NULL              |                             |
+---------------+------------------+------+-----+-------------------+-----------------------------+
```
### oauth_scopes
```
+------------+-------------+------+-----+---------+-------+
| Field      | Type        | Null | Key | Default | Extra |
+------------+-------------+------+-----+---------+-------+
| scope      | varchar(80) | NO   | PRI | NULL    |       |
| is_default | tinyint(1)  | YES  |     | NULL    |       |
+------------+-------------+------+-----+---------+-------+
```
### oauth_users
```
+----------------+------------------+------+-----+---------+----------------+
| Field          | Type             | Null | Key | Default | Extra          |
+----------------+------------------+------+-----+---------+----------------+
| username       | varchar(80)      | YES  |     | NULL    |                |
| password       | varchar(80)      | YES  |     | NULL    |                |
| first_name     | varchar(80)      | YES  |     | NULL    |                |
| last_name      | varchar(80)      | YES  |     | NULL    |                |
| email          | varchar(2000)    | YES  |     | NULL    |                |
| email_verified | tinyint(1)       | YES  |     | NULL    |                |
| scope          | varchar(4000)    | YES  |     | NULL    |                |
+----------------+------------------+------+-----+---------+----------------+
```

## Describe alternatives you've considered
I can't say that I'm fully satisfied with mentioned schema. My complaints are:
* `oauth_users`.`password` column limited to 80 characters which can be not enough for future password encryption algorithms. In PHP docs [`password_hash`](https://www.php.net/manual/en/function.password-hash.php) recommended length is 255.
* Schema doesn't set column collations. It's important for case-sensitive tokens(base64 e.g.). More info in submitted [issue](https://github.com/dsquier/oauth2-server-php-mysql/issues/12)
* It's not obvious that `oauth_users`.`username` is `user_id` mentioned in other tables. Maybe column comment can make it more clear. 

**If there are database architects in our community, they can significantly improve this schema.**

### PR DESCRIPTION
**Pull request by ybelenko**
Database schema copied from [oauth2-server-php-mysql](https://github.com/dsquier/oauth2-server-php-mysql/) repo.

I can't say that I'm fully satisfied with mentioned schema. My complaints are:
* ~`oauth_users`.`password` column limited to 80 characters which can be not enough for future password encryption algorithms. In PHP docs [`password_hash`](https://www.php.net/manual/en/function.password-hash.php) recommended length is 255.~
* Schema doesn't set column collations. It's important for case-sensitive tokens(base64 e.g.). More info in submitted [issue](https://github.com/dsquier/oauth2-server-php-mysql/issues/12)
* It's not obvious that `oauth_users`.`username` is `user_id` mentioned in other tables. Maybe column comment can make it more clear.

Tables structure described in related issue.

Closes #3550

@jimschubert is welcome to review or throw an opinion.

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by jimschubert** [comment]
If this schema is copied from another project, it should include appropriate attribution.  I don't see a license on the linked repo, so you'll want to reach out to that user before we can integrate this. The license should be compatible with Apache 2.0.

File: modules/openapi-generator/src/main/resources/mysql-schema/mysql_schema.mustache [hunk 1]
```diff
@@ -42,4 +42,108 @@ CREATE TABLE IF NOT EXISTS {{#defaultDatabaseName}}`{{{defaultDatabaseName}}}`.{
...
+  `scope`               VARCHAR(4000)  DEFAULT NULL,
+  PRIMARY KEY (`refresh_token`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+--
+-- Table structure for table `oauth_users`
+--
+CREATE TABLE IF NOT EXISTS `oauth_users` (
+  `username`            VARCHAR(80)    DEFAULT NULL,
+  `password`            VARCHAR(80)    DEFAULT NULL,
```
Comment (jimschubert) [inline modules/openapi-generator/src/main/resources/mysql-schema/mysql_schema.mustache:103]: a password shouldn't be stored in the database. Really, we should have a hashed password or an encrypted blob

Comment (ybelenko) [reply modules/openapi-generator/src/main/resources/mysql-schema/mysql_schema.mustache:103]: Do you suggest to rename column into `password_hash` to make it more clear? I thought it's obvious for any anyone. I'm more worried about small column length (80).

**Comment by ybelenko** [comment]
> If this schema is copied from another project, it should include appropriate attribution. I don't see a license on the linked repo, so you'll want to reach out to that user before we can integrate this. The license should be compatible with Apache 2.0.

There is only one maintainer(@dsquier) of the repo and his profile without email or any contacts. I've left [3 issues](https://github.com/dsquier/oauth2-server-php-mysql/issues/created_by/ybelenko) in Aug 1, 2019. No reaction.

I've noticed the same schema at another place [bshaffer/oauth2-server-php](https://github.com/bshaffer/oauth2-server-php/blob/5a0c8000d4763b276919e2106f54eddda6bc50fa/src/OAuth2/Storage/Pdo.php#L649-L727) maybe it can help us somehow? I'm not very experienced in a license related things.

**Comment by dsquier** [comment]
I've updated the https://github.com/dsquier/oauth2-server-php-mysql repo to include the license, which is Apache 2.0. Hopefully this aligns well with your project. Good luck!

**Pull request merged**